A few types can't be found

I'm getting an error on the `EventHandlerNonNull` and `BinaryType` types.

```
ERROR in node_modules/mock-socket/index.d.ts(21,13): error TS2304: Cannot find name 'EventHandlerNonNull'.
node_modules/mock-socket/index.d.ts(22,14): error TS2304: Cannot find name 'EventHandlerNonNull'.
node_modules/mock-socket/index.d.ts(23,14): error TS2304: Cannot find name 'EventHandlerNonNull'.
node_modules/mock-socket/index.d.ts(28,16): error TS2304: Cannot find name 'EventHandlerNonNull'.
node_modules/mock-socket/index.d.ts(29,17): error TS2304: Cannot find name 'BinaryType'.
```

I already have the `dom` library defined in my `tsconfig.json` file, and this is where those types seem to be be.

I'm using version 8.0.2.

How can this be resolved? I have tried many things, to no avail.

--- a/index.d.ts
+++ b/index.d.ts
@@ -17,6 +17,33 @@ declare module 'mock-socket' {
     open: () => void;
   }
 
+  interface OpenEvent {
+    type: string;
+    target: WebSocket;
+  }
+
+  interface ErrorEvent {
+    error: any;
+    message: string;
+    type: string;
+    target: WebSocket;
+  }
+
+  interface CloseEvent {
+    wasClean: boolean;
+    code: number;
+    reason: string;
+    type: string;
+    target: WebSocket;
+  }
+
+  type Data = string | Buffer | ArrayBuffer | Buffer[];
+  interface MessageEvent {
+    data: Data;
+    type: string;
+    target: WebSocket;
+  }
+
   //
   // https://html.spec.whatwg.org/multipage/web-sockets.html#websocket
   //
@@ -37,14 +64,14 @@ declare module 'mock-socket' {
     readonly readyState: number;
     readonly bufferedAmount: number;
 
-    onopen: EventHandlerNonNull;
-    onerror: EventHandlerNonNull;
-    onclose: EventHandlerNonNull;
+    onopen: (event: OpenEvent) => void;
+    onerror: (event: ErrorEvent) => void;
+    onclose: (event: CloseEvent) => void;
+    onmessage: (event: MessageEvent) => void;
     readonly extensions: string;
     readonly protocol: string;
     close(code?: number, reason?: string): void;
 
-    onmessage: EventHandlerNonNull;
     binaryType: BinaryType;
     send(data: string | Blob | ArrayBuffer | ArrayBufferView): void;
     on<K extends keyof WebSocketCallbackMap>(type: K, callback: WebSocketCallbackMap[K]): void;

